src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -297,7 +298,9 @@
   []
   (map->Query {:projections {"logs" {:type :json
                                      :queryable? false
-                                     :field :reports.logs}
+                                     :field (h/coalesce :logs
```

Comment:
I think we also need tests around the various scenarios here (data in JSON, data in JSONB etc)